### PR TITLE
Advanced SMOS filters

### PR DESCRIPTION
**Pull request by awst-baum**
This PR implements QA4SM-281: it uses the new AdvancedMaskingAdapter to implement the more sophisticated SMOS filters. The filters still need to be checked for correctness.
**This PR needs pytesmo PR https://github.com/TUW-GEO/pytesmo/pull/190 to work**

### CODE REVIEW

**Comment by wpreimes** [comment]
What is the current selection of filters based on? I think we should include all the flags.
- Quality Recommended, on by default : Single masking adapter that filters by ``'quality_flag'==int(0)``
and if this one is not selected (if it is selected the other filters shouldn't be available to select):
- Exclude moderate topo: Filter by ``scene_flag != 0b00000001``
- Exclude strong topo: Filter by  ``scene_flag != 0b00000010``
- Exclude polluted scene: Filter by ``scene_flag  != 0b00000100``
- Exclude frozen scenes: Filter by ``scene_flag  != 0b00001000``
- Exclude high RMSE areas: Filter by ``processing_flag  != 0b00000001``

**Comment by awst-baum** [comment]
> What is the current selection of filters based on?

Tracy's preferences? :D

> I think we should include all the flags.

On it. Can you please provide some help text for 'Exclude moderate topo' and 'Exclude strong topo'? What's moderate topo? Almwiesen? ;)

File: validator/fixtures/filters.json [hunk 1]
```diff
@@ -202,5 +202,41 @@
...
             "default_parameter": "AMMA-CATCH,BIEBRZA-S-1,BNZ-LTER,CARBOAFRICA,COSMOS,CTP-SMTMN,DAHRA,FLUXNET-AMERIFLUX,FMI,FR-Aqui,HOBE,iRON,LAB-net,MySMNet,OZNET,PBO-H2O,REMEDHUS,RISMA,RSMN,SCAN,SMOSMANIA,TERENO,USCRN,WEGENERNET,WSMN"
         }
+    },
+    {
+        "model": "validator.datafilter",
+        "pk": 19,
+        "fields": {
+            "name": "FIL_SMOS_UNFROZEN",
+            "description": "Soil not frozen",
+            "help_text": "Scene_Flags != 00001000 and Soil_Temperature_Level1 >= 273",
```
[the remark this answers is not in the document]
Comment (awst-baum) [reply validator/fixtures/filters.json]: Changed the wording

File: validator/fixtures/filters.json [hunk 2]
```diff
@@ -202,5 +202,41 @@
...
+            "dialog_name": null,
+            "default_parameter": ""
+        }
+    },
+    {
+        "model": "validator.datafilter",
+        "pk": 20,
+        "fields": {
+            "name": "FIL_SMOS_UNPOLLUTED",
+            "description": "Scene not polluted",
```
[the remark this answers is not in the document]
Comment (awst-baum) [reply validator/fixtures/filters.json:223]: Yep, changed it.

File: validator/validation/filters.py [hunk 3]
```diff
@@ -107,78 +137,96 @@ def setup_filtering(reader, filters, param_filters, dataset, variable):
...
             continue
 
         if(fil.name == "FIL_SMOS_QUAL_RECOMMENDED"):
-            filtered_reader = SelfMaskingAdapter(filtered_reader, '==', 0, 'Quality_Flag')
+            masking_filters.append( ('Quality_Flag', '==', 0) )
+            continue
+
+        if(fil.name == "FIL_SMOS_UNFROZEN"):
+            masking_filters.append( ('Scene_Flags', smos_exclude_bitmask, 0b00001000) )
+            masking_filters.append( ('Soil_Temperature_Level1', '>=', 273) )
```
[the remark this answers is not in the document]
Comment (awst-baum) [reply validator/validation/filters.py]: I've removed the second filter now.

**Pull request merged**